client/src/pages/AccountsPage/AccountsPageContext.tsx

```diff
@@ -0,0 +1,101 @@
+import { useState } from 'react';
+import { DataItem } from 'src/components/common/DataTile';
+import { useGetData } from 'src/hooks/useGetData';
+import { usePaginationData, UsePaginationDefaultProps } from 'src/hooks/usePaginationData';
+import {
+  getCurrentTokenSupply,
+  getTokenCirculatingSupply,
+  getStaking,
+  StakingResponse,
+  getLockedTokens,
+} from 'src/services/fio-protocol';
+import { getROE } from 'src/services/roe';
+import { getAccounts } from 'src/services/accounts';
+import { formatTokenValue } from 'src/utils/general';
+import { Account, AccountSortOption } from '@shared/types/accounts';
+
+
+type UseAccountsPageType = {
+  loading: boolean;
+  stats: DataItem[];
+  accounts: Account[];
+  accountsLoading: boolean;
+  totalAccounts: number;
+  sort: AccountSortOption;
+  setSort: (option: AccountSortOption) => void;
+  paginationProps: UsePaginationDefaultProps;
+};
+
+export const useAccountsPageContext = (): UseAccountsPageType => {
+  const [sort, setSort] = useState<AccountSortOption>('pk_account_id');
+
+  const { response: currentSupply, loading: currentSupplyLoading } = useGetData<number>({
+    action: getCurrentTokenSupply,
+  });
+  const { response: circulatingSupply, loading: circulatingSupplyLoading } = useGetData<number>({
+    action: getTokenCirculatingSupply,
+  });
+  const { response: roe, loading: roeLoading } = useGetData<string>({ action: getROE });
+  const { response: stakedFIO, loading: stakedFIOLoading } = useGetData<StakingResponse>({
+    action: getStaking,
+  });
+  const { response: lockedSupply, loading: lockedSupplyLoading } = useGetData<number>({
+    action: getLockedTokens,
+  });
+
+  const {
+    data: accounts,
+    loading: accountsLoading,
+    total,
+    ...paginationProps
+  } = usePaginationData<Account>({
+    action: getAccounts,
+    params: {
+      sort,
+    },
+    returnTotal: true,
+  });
+
+  const stats: DataItem[] = [
+    {
+      title: 'Max Token Supply',
+      value: '1,000,000,000 FIO',
+    },
+    {
+      title: 'FIO Price',
+      value: `${formatTokenValue(roe)} USD`,
+    },
+    {
+      title: 'Current Token Supply',
+      value: `${formatTokenValue(currentSupply)} FIO`,
+    },
+    {
+      title: 'Circulating Token Supply',
+      value: `${formatTokenValue(circulatingSupply)} FIO`,
+    },
+    {
+      title: 'Staked FIO',
+      value: `${formatTokenValue(stakedFIO?.staked_token_pool)} FIO`,
+    },
+    {
+      title: 'Locked Supply',
+      value: `${formatTokenValue(lockedSupply)} FIO`,
+    },
+  ];
```

Comment:
does it makes sense to create one more service which would return this stats array and unite getLockedTokens, getStaking, getROE etc?